lib/DxcSupport/Unicode.cpp

```diff
@@ -28,7 +28,9 @@ int MultiByteToWideChar(uint32_t /*CodePage*/, uint32_t /*dwFlags*/,
                         const char *lpMultiByteStr, int cbMultiByte,
                         wchar_t *lpWideCharStr, int cchWideChar) {
 
-  if (cbMultiByte == 0) {
+  // Check for invalid sizes or potential overflow.
+  if (cbMultiByte == 0 || cbMultiByte < -1 || cbMultiByte > (INT32_MAX - 1) ||
+      cchWideChar < 0 || cchWideChar > (INT32_MAX - 1)) {
```

Comment:
Isn't `cbMultiByte > (INT32_MAX - 1)` equivalent to `cbMultiByte == INT32_MAX`? Unless we were to change `cbMultiByte` and `cchWideChar` to larger types I think the equality check is clearer, no?